src/types/queries.rs

```diff
@@ -24,6 +24,7 @@ use std::net::IpAddr;
     derive_more::Display,
 )]
 #[display(fmt = "P{_0}")]
+#[serde(transparent)]
```

Comment:
This is changing the serde implementation to match `ProtocolVersion`, since this type is new and not released it should be fine.